packages/ember-debug/lib/handlers.js

```diff
@@ -1,9 +1,23 @@
 import isPlainFunction from 'ember-debug/is-plain-function';
+import deprecate from 'ember-debug/deprecate';
 
 export let HANDLERS = { };
+export let testAsFunctionDeprecation = 'Functional arguments are ambiguous with constructors. ' +
```

Comment:
I think that we need to mention what they did that triggered the message. It makes this slightly annoying (due to passing through what was called), but I think it will make it much easier to track down.

One possible solution might be:

```javascript
export function generateTestAsFunctionDeprecation(source) {
  return `Calling \`${source}\` with a function argument is deprecated. Please ` +
    `use \`!!Constructor\` for constructors, or an `IIFE` to compute the test for deprecation. ` +
    `In a future version functions will be treated as truthy values instead of being executed.`
}
```

Then you would have to pass through `'Ember.' + type` in `invoke` function below (and thread that through to `normalizeTest`).